app/src/main/java/com/kickstarter/ui/adapters/RewardAndAddOnsAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.kickstarter.ui.adapters
+
+import android.util.Pair
+import android.view.View
+import com.kickstarter.R
+import com.kickstarter.models.Reward
+import com.kickstarter.ui.data.ProjectData
+import com.kickstarter.ui.viewholders.*
+
+class RewardAndAddOnsAdapter(private val viewListener: RewardAndAddOnsAdapter.ViewListener) : KSAdapter() {
+
+    interface ViewListener:RewardViewHolder.Delegate, AddOnViewHolder.ViewListener
+
+    init {
+        insertSection(SECTION_REWARD_CARD, emptyList<Reward>())
+        insertSection(SECTION_ADD_ONS_CARD, emptyList<Reward>())
+    }
+
+    override fun layout(sectionRow: SectionRow): Int = when (sectionRow.section()){
+        SECTION_REWARD_CARD -> R.layout.item_reward
+        SECTION_ADD_ONS_CARD -> R.layout.item_add_on
+        else -> 0
+    }
+
+    override fun viewHolder(layout: Int, view: View): KSViewHolder {
+        return when(layout) {
+            R.layout.item_reward -> RewardViewHolder(view, viewListener)
+            R.layout.item_add_on -> AddOnViewHolder(view, viewListener)
+            else -> EmptyViewHolder(view)
+        }
+    }
+
+    fun populateDataForAddOns(rewards: List<Pair<ProjectData, Reward>>) {
+        setSection(SECTION_ADD_ONS_CARD, rewards)
+        notifyDataSetChanged()
+    }
+
+    fun populateDataForRewad(reward: Pair<ProjectData, Reward>) {
+        setSection(SECTION_REWARD_CARD, listOf(reward))
+        notifyDataSetChanged()
+    }
+
+    companion object {
```

Comment:
Random question, is there a standard for where the companion object should be placed in the file? For example it should always be placed toward the top or bottom of the file?

Reply:
I'be mostly see it on the bottom of the classes -> [Kotlin code conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
<img width="545" alt="Screen Shot 2020-06-05 at 1 32 32 PM" src="https://user-images.githubusercontent.com/4083656/83920445-16719500-a731-11ea-8e1b-48e3f5ce19e0.png">